Source/WebCore/platform/Timer.cpp

```diff
@@ -280,6 +284,9 @@ static_assert(sizeof(DeferrableOneShotTimer) == sizeof(SameSizeAsDeferrableOneSh
 
 TimerBase::TimerBase()
 {
+#if USE(WEB_THREAD)
+    ELEASE_ASSERT(WebThreadIsLockedOrDisabled());
+#endif
```

Comment:
If I remember correctly, this check is too simplistic because:
- You must only perform this check in the UI process, not in any `com.apple.WebKit.*` processes.
- The timer is valid if the WebThread is disabled or if `(WebThreadIsCurrent() || WebThreadIsLocked())`, not just if `WebThreadIsLockedOrDisabled()`.

See this bug and the last PR posted:

Bug [185330](https://bugs.webkit.org/show_bug.cgi?id=185330): [iOS] Add assert to catch improper use of WebCore::Timer in UI Process
- Patch v5:  <https://bugs.webkit.org/attachment.cgi?id=344224&action=review>

As I recall, Patch v5 still wasn't quite correct (don't remember the details), and then I ran out of time to continue working on it.

Reply:
FWIW, the web thread should _never_ be enabled in the web process in the first place.

Reply:
> FWIW, the web thread should _never_ be enabled in the web process in the first place (or any of the other WebKit child XPC services)

Right. `WebThreadIsLockedOrDisabled` should return true there because the web thread is never enabled in those processes.

Reply:
> * You must only perform this check in the UI process, not in any `com.apple.WebKit.*` processes.
> * The timer is valid if the WebThread is disabled or if `(WebThreadIsCurrent() || WebThreadIsLocked())`, not just if `WebThreadIsLockedOrDisabled()`.

WebThreadIsLocked() returns true when WebThreadIsCurrent() is true.